crates/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -483,9 +488,7 @@ impl<'dfg> InsertInstructionResult<'dfg> {
             InsertInstructionResult::Results(results) => Cow::Borrowed(results),
             InsertInstructionResult::SimplifiedTo(result) => Cow::Owned(vec![result]),
             InsertInstructionResult::SimplifiedToMultiple(results) => Cow::Owned(results),
-            InsertInstructionResult::InstructionRemoved => {
-                panic!("InsertInstructionResult::results called on a removed instruction")
-            }
+            InsertInstructionResult::InstructionRemoved => Cow::Owned(vec![]),
```

Comment:
This was changed because it was causing panics when `assert_constant` calls were simplified out immediately after inserting them